pdm/cli/commands/run.py

```diff
@@ -223,25 +228,38 @@ def _run_task(
         )
 
     def run(
-        self, command: str, args: Sequence[str], opts: TaskOptions | None = None
+        self, command: str, args: list[str], opts: TaskOptions | None = None
     ) -> int:
+        if command in self.hooks.skip:
+            return 0
         task = self._get_task(command)
         if task is not None:
+
             pre_task = self._get_task(f"pre_{command}")
-            if pre_task is not None:
+            if (
+                pre_task is not None
+                and not self.hooks.skip_all
+                and not self.hooks.skip_pre
+                and pre_task.name not in self.hooks.skip
+            ):
```

Comment:
I love the abstraction of `HookManager` but these lines seem to be repeating. What about adding a method `HookManager.should_skip(name: str) -> bool` to check for that?

Reply:
I've gone for `HookManager.should_run(name: str) -> bool` as it avoids double negations.
It is more easy to read.